R/conclass.R

```diff
@@ -530,23 +572,41 @@ Conos <- R6::R6Class("Conos", lock_objects=F,
     #' @description Plot joint graph.
     #'
     #' @param clustering name of the clustering to use
+    #' @param embedding.type name of the embedding to use
     #' @param groups a factor on cells to use for coloring.
     #' @param colors a color factor (named with cell names) use for cell coloring.
     #' @param gene show expression of a gene.
     #' @param subset a subset of cells to show.
-    plotGraph=function(color.by='cluster', clustering=NULL, groups=NULL, colors=NULL, gene=NULL, plot.theme=NULL, subset=NULL, ...) {
-      if(is.null(self$embedding)) {
+    plotGraph=function(color.by='cluster', clustering=NULL, embedding.type=NULL, groups=NULL, colors=NULL, gene=NULL, plot.theme=NULL, subset=NULL, ...) {
+      if (length(self$embeddings) == 0) {
         self$embedGraph()
       }
+      
+      if (!is.null(embedding.type)){
+        ## check embedding exists in list
+        if (embedding.type %in% names(self$embeddings)){
+          emb <- self$embeddings[[embedding.type]]
+        } else{
+          stop(paste("Embedding", embedding.type, "is not found in any of the joint embeddings"))
+        }
+      } else{
+        ## grab last element in embeddings list
+        emb <- self$embeddings[length(self$embeddings)]
+      }
 
-      emb <- self$embedding;
-      if(!is.null(subset)) {
-        emb <- emb[rownames(emb) %in% subset,,drop=F]
+      if (!is.null(subset)) {
+        emb <- emb[rownames(emb) %in% subset,,drop=FALSE]
+      }
+      ## catch for deprecated 'embedding'
+      if ( !is.null(subset) && !is.null(embedding) ){
```

Comment:
Perhaps, you've meant `self$embedding` here. Though I'm not sure if we need this `if` at all. Maybe we should remove lines 601-604.

Reply:
Yes, I think this was a mistake due to a conceptual misunderstand; let's regroup below.